stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MediaAttachmentsGroupView.kt

```diff
@@ -69,60 +69,66 @@ internal class ImageAttachmentsGroupView : ConstraintLayout {
-                images[1],
-                images[2],
-                images[3],
-                images.size - MAX_PREVIEW_COUNT
+                media.first(),
+                media[1],
+                media[2],
+                media[3],
+                media.size - MAX_PREVIEW_COUNT
             )
         }
-        (background as? MaterialShapeDrawable)?.shapeAppearanceModel?.let(::applyToImages)
+        (background as? MaterialShapeDrawable)?.shapeAppearanceModel?.let(::applyToMediaPreviews)
     }
 
     private fun showOne(first: Attachment) {
         removeAllViews()
-        val imageAttachmentView = createImageAttachmentView()
-        addView(imageAttachmentView)
-        state = State.OneView(imageAttachmentView)
+        val mediaAttachmentView = createMediaAttachmentView()
+        addView(mediaAttachmentView)
+        state = State.OneView(mediaAttachmentView)
         ConstraintSet().apply {
-            constrainMaxHeight(imageAttachmentView.id, maxImageAttachmentHeight)
-            constrainWidth(imageAttachmentView.id, ViewGroup.LayoutParams.MATCH_PARENT)
-            constrainViewToParentBySide(imageAttachmentView, ConstraintSet.LEFT)
-            constrainViewToParentBySide(imageAttachmentView, ConstraintSet.RIGHT)
-            constrainViewToParentBySide(imageAttachmentView, ConstraintSet.TOP)
-            constrainViewToParentBySide(imageAttachmentView, ConstraintSet.BOTTOM)
+            constrainMaxHeight(mediaAttachmentView.id, maxMediaAttachmentHeight)
+            constrainWidth(mediaAttachmentView.id, ViewGroup.LayoutParams.MATCH_PARENT)
+            constrainViewToParentBySide(mediaAttachmentView, ConstraintSet.LEFT)
+            constrainViewToParentBySide(mediaAttachmentView, ConstraintSet.RIGHT)
+            constrainViewToParentBySide(mediaAttachmentView, ConstraintSet.TOP)
+            constrainViewToParentBySide(mediaAttachmentView, ConstraintSet.BOTTOM)
 
-            val imageWidth = first.originalWidth?.toFloat()
-            val imageHeight = first.originalHeight?.toFloat()
+            val mediaWidth = first.originalWidth?.toFloat()
+            val mediaHeight = first.originalHeight?.toFloat()
 
-            // Used to set a dimension ratio before we load an image
+            // Used to set a dimension ratio before we load the media
             // so that message positions don't jump after we load it.
-            if (imageWidth != null && imageHeight != null) {
-                val ratio = (imageWidth / imageHeight).toString()
-                this.setDimensionRatio(imageAttachmentView.id, ratio)
+            if (mediaWidth != null && mediaHeight != null) {
+                val ratio = (mediaWidth / mediaHeight).toString()
+                this.setDimensionRatio(mediaAttachmentView.id, ratio)
             } else {
-                constrainHeight(imageAttachmentView.id, LayoutParams.WRAP_CONTENT)
+                this.setDimensionRatio(mediaAttachmentView.id, "1")
```

Comment:
Previously we let attachments without dimensions wrap height.
This would still incur jumping, but would allow us to go forwards in a non-breaking way.

Since we are now releasing a major, I've decided to use a fixed square ratio for images without dimension. The new CDN roll-out should mean that this is mostly a case for older attachments and should not affect new ones.